zerocopy-derive/src/lib.rs

```diff
@@ -195,16 +195,20 @@ fn derive_from_bytes_union(ast: &DeriveInput, unn: &DataUnion) -> proc_macro2::T
 // - `repr(packed)`
 
 fn derive_as_bytes_struct(ast: &DeriveInput, strct: &DataStruct) -> proc_macro2::TokenStream {
-    // TODO(#10): Support type parameters.
-    if !ast.generics.params.is_empty() {
+    let reprs = try_or_print!(STRUCT_UNION_AS_BYTES_CFG.validate_reprs(ast));
+    let is_transparent = reprs.contains(&StructRepr::Transparent);
+
+    // TODO(#10): Support type parameters for non-transparent structs.
+    if !ast.generics.params.is_empty() && !is_transparent {
         return Error::new(Span::call_site(), "unsupported on types with type parameters")
             .to_compile_error();
```

Comment:
Maybe change this to "unsupported on non-repr(transparent) types with type parameters" or something similar?

Reply:
Done